widget/embedded/src/hooks/useConfirmSwap/useConfirmSwap.ts

```diff
@@ -63,70 +51,64 @@ export function useConfirmSwap(): ConfirmSwap {
       };
     }
 
-    const requestBody = createQuoteRequestBody({
-      fromToken,
-      toToken,
-      inputAmount,
-      wallets: connectedWallets,
-      selectedWallets,
-      excludeLiquiditySources: enableNewLiquiditySources,
-      disabledLiquiditySources,
-      slippage: userSlippage,
-      affiliateRef,
-      affiliatePercent,
-      affiliateWallets,
-      initialQuote: initialQuote ?? undefined,
+    const wallets = selectedWallets.reduce((acc, item) => {
+      acc[item.chain] = item.address;
+      return acc;
+    }, {} as Record<string, string>);
+    const requestBody: ConfirmRouteRequest = {
+      requestId: initialQuote?.requestId || '',
+      selectedWallets: wallets,
       destination: customDestination,
-    });
-    if (isFeatureEnabled('experimentalRoute', features)) {
-      requestBody.experimental = true;
-    }
-    let currentQuote: BestRouteResponse;
+    };
+
     try {
-      currentQuote = await fetchQuote(requestBody).then((response) =>
-        throwErrorIfResponseIsNotValid(response)
-      );
-      setQuote(currentQuote);
+      return await fetchQuote(requestBody).then((response) => {
+        const { result: currentQuote } = response;
+        throwErrorIfResponseIsNotValid({
+          diagnosisMessages: currentQuote.diagnosisMessages,
+          requestId: currentQuote.requestId,
+          swaps: currentQuote.result?.swaps,
+        });
```

Comment:
Consider verifying the presence of the result value in the response. Refer to the documentation for "ConfirmRouteResponse" for more details.